tracer/src/Datadog.Trace/PDBs/DatadogPdbReader.cs

```diff
@@ -76,12 +76,15 @@ private static bool TryFindPdbFile(Assembly assembly, [NotNullWhen(true)] out st
             // and the PDB may not be copied to this folder until later or never, as it is only copied on demand (usually when an exception is thrown).
             // In these cases, we'll try to find it in the application directory (e.g. C:\inetpub\wwwroot\MyApp\bin\MyApp.pdb).
             string fileName = Path.GetFileName(pdbInSameFolder);
-            var pdbInAppDirectory = Directory.EnumerateFiles(System.Web.Hosting.HostingEnvironment.ApplicationPhysicalPath, fileName, SearchOption.AllDirectories).FirstOrDefault();
-
-            if (pdbInAppDirectory != null)
+            string applicationDirectory = System.Web.Hosting.HostingEnvironment.ApplicationPhysicalPath;
+            if (applicationDirectory != null)
             {
-                pdbFullPath = pdbInAppDirectory;
-                return true;
+                var pdbInAppDirectory = Directory.EnumerateFiles(applicationDirectory, fileName, SearchOption.AllDirectories).FirstOrDefault();
```

Comment:
Given that this also could throw if applicationDirectory is whitespace, has invalid chars, doesn't exist, or doesn't have permission to access, maybe we should just wrap in a try-catch to be safe?

Reply:
Great point! Done in cdef657667d32a4db377ba0e28915781da3c8cfb